src/wasm/wasm.cpp

```diff
@@ -578,25 +578,28 @@ bool Function::isVar(Index index) {
 }
 
 bool Function::hasLocalName(Index index) const {
-  return index < localNames.size() && localNames[index].is();
+  return localNames.find(index) != localNames.end();
 }
 
 Name Function::getLocalName(Index index) {
-  assert(hasLocalName(index));
-  return localNames[index];
+  auto nameIt = localNames.find(index);
```

Comment:
i think we can do `localNames.at(index)` which both does the check+returns the value.